src/gui/EditWidgetIcons.cpp

```diff
@@ -314,43 +317,76 @@ void EditWidgetIcons::addCustomIconFromFile()
         QString filter = QString("%1 (%2);;%3 (*)").arg(tr("Images"),
                     Tools::imageReaderFilter(), tr("All files"));
 
-        QString filename = QFileDialog::getOpenFileName(
-                    this, tr("Select Image"), "", filter);
-        if (!filename.isEmpty()) {
-            auto icon = QImage(filename);
-            if (!icon.isNull()) {
-                addCustomIcon(QImage(filename));
+        auto filenames = QFileDialog::getOpenFileNames(this, tr("Select Image(s)"), "", filter);
+        if (!filenames.empty()) {
+            QStringList errornames;
+            int numexisting = 0;
+            for (const auto& filename : filenames) {
+                if (!filename.isEmpty()) {
+                    auto icon = QImage(filename);
+                    if (!icon.isNull()) {
+                        if (!addCustomIcon(QImage(filename))) {
```

Comment:
Combine conditions.

Reply:
They have to be separate because I want to fall through (ie, not an error).

Reply:
That would be `else if (icon.isNull())`.